tools/dh_eoscontent

```diff
@@ -111,20 +111,30 @@ sub app_appdata_file {
 	my $tmp=shift;
 
 	# See if this package has the appdata file
-	my $appdata = $tmp . '/usr/share/appdata/' . $app_id
-	    . '.appdata.xml';
-	verbose_print("Checking for app-data file $appdata");
-	if (-l $appdata) {
-		my $link = readlink($appdata);
-		if ($link =~ /^\//) {
-			# If the symlink is absolute, make it relative
-			# to the tmpdir
-			$appdata = $tmp . $link;
-		} else {
-			# Resolve it to an absolute path
-			$appdata = abs_path($appdata);
+	my $appdata;
+	my @appdata_dirs = ('metainfo', 'appdata');
+	foreach my $appdata_dir (@{appdata_dirs}) {
+		my $appdata_file = $tmp . '/usr/share/' . $appdata_dir . '/' . $app_id
+		    . '.appdata.xml';
+		verbose_print("Checking for app-data file $appdata_file");
+
+		next unless (-e $appdata_file);
+		$appdata = $appdata_file;
+
+		if (-l $appdata) {
+			my $link = readlink($appdata);
+			if ($link =~ /^\//) {
+				# If the symlink is absolute, make it relative
+				# to the tmpdir
+				$appdata = $tmp . $link;
+			} else {
+				# Resolve it to an absolute path
+				$appdata = abs_path($appdata);
+			}
 		}
+
 		verbose_print("Resolved symlink to $appdata");
+		last;
```

Comment:
Minor suggestion - Instead of breaking from the loop, just `return $appdata` here and drop the `return` below. If nothing is found, the result is the same - the function completes without explicitly returning (in my limited perl understanding, that makes the return value `undef`). Then you could also move the `my $appdata` declaration inside this block.

Reply:
Sounds good to me. It's the first time I ever wrote anything in Perl, so I was unsure what the result would be if the function didn't return anything, that's why I did it this way: to have a clear and only way to return from.

I'm happy to change it if that's ok, though, I won't be the one complaining on Perl!